component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java

```diff
@@ -517,5 +527,53 @@ private void addMatchStringsToList(Matcher matcher, Set<String> matches) {
         }
     }
 
+    private String getUserStoreDomainName(String userIdentifier, AuthenticationContext authenticationContext)
+            throws UserStoreException, AuthenticationFailedException {
+        if (Boolean.valueOf(getAuthenticatorConfig().getParameterMap().get(X509CertificateConstants.SEARCH_ALL_USERSTORES))) {
+            String[] filteredUsers = X509CertificateUtil.getUserRealm(userIdentifier).getUserStoreManager().listUsers
+                    (MultitenantUtils.getTenantAwareUsername(userIdentifier), X509CertificateConstants.MAX_ITEM_LIMIT_UNLIMITED);
+            if (filteredUsers.length == 1) {
+                if (log.isDebugEnabled()) {
+                    log.debug("User exists with the user name: " + userIdentifier);
+                }
+                if (filteredUsers[0].indexOf("/") > 0) {
+                    String[] subjectIdentifierSplits = filteredUsers[0].split("/", 2);
+                    return subjectIdentifierSplits[0];
+                } else {
+                    return null;
+                }
+            } else if (filteredUsers.length > 1) {
+                authenticationContext.setProperty(X509CertificateConstants.X509_CERTIFICATE_ERROR_CODE,
+                        X509CertificateConstants.USERNAME_CONFLICT);
+                throw new AuthenticationFailedException("Conflicting users with user name: " + userIdentifier);
+            } else if (getAuthenticatorConfig().getParameterMap().containsKey(X509CertificateConstants.LOGIN_CLAIM_URIS)) {
+                for (String multiAttributeClaimUri : getAuthenticatorConfig().getParameterMap()
```

Comment:
Shall we assign getAuthenticatorConfig().getParameterMap()
                        .get(X509CertificateConstants.LOGIN_CLAIM_URIS).split(",") to an array first?